encoders/src/main/scala/scala3encoders/derivation/Deserializer.scala

```diff
@@ -22,20 +22,14 @@ trait Deserializer[T]:
   def nullable: Boolean = true
 
 object Deserializer:
-
-  // see ScalaReflection.deserializerFor
```

Comment:
`ScalaReflection.deserializerFor` is gone in latest versions of Spark so this is outdated now

Reply:
it's actually still there:
https://github.com/apache/spark/blob/v3.5.0/sql/api/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala#L242

Reply:
Would it make sense to keep the comment but update it?

Reply:
It would make sense (I hope) ;-)